### ISSUE TITLE
time series forecasting using rnn

### ISSUE THREAD

**Issue by bifeng** [apache/incubator-mxnet#3572]
`
library(Quandl)
sh_stock_ex <- Quandl("YAHOO/SS_600292", type="xts")
library(xts)
data <- scale(sh_stock_ex[-1,5])  
feat <- merge(na.trim(lag(data,1)), na.trim(lag(data,2)), na.trim(lag(data,3)), na.trim(lag(data,4)), 
              na.trim(lag(data,5)), all=FALSE)

dataset <- merge(feat, data, all = FALSE)
colnames(dataset) <- c("lag.1", "lag.2","lag.3","lag.4","lag.5", "obj")

index <- 1:4000
training <- as.data.frame(dataset[index,])
testing <- as.data.frame(dataset[-index,])

library(mxnet)
train.x <- data.matrix(training[,-6])
train.y <- training[,6]
test.x <- data.matrix(testing[,-6])
test.y <- testing[,6]

get.label <- function(X) {
  label <- array(0, dim=dim(X))
  d <- dim(X)[1]
  w <- dim(X)[2]
  for (i in 0:(w-1)) {
    for (j in 1:d) {
      label[i_d+j] <- X[(i_d+j)%%(w*d)+1]
    }
  }
  return (label)
}
X.train.label <- get.label(t(train.x))
X.val.label <- get.label(t(test.x))

X.train <- list(data=t(train.x), label=X.train.label)
X.val <- list(data=t(test.x), label=X.val.label)

batch.size = 5
seq.len = 5  
num.hidden = 5
num.embed = 5
num.rnn.layer = 1
num.round = 100
update.period = 1
learning.rate= 0.1
wd=0.00001
clip_gradient=1

mx.set.seed(0)
model <- mx.rnn(X.train, X.val, num.rnn.layer=num.rnn.layer, seq.len=seq.len, num.hidden=num.hidden,
                num.embed=num.embed, num.label=5, batch.size=batch.size, input.size=5, ctx = mx.cpu(),
                num.round = num.round, update.period = update.period, initializer = mx.init.uniform(0.01),
                dropout = 0, optimizer = "sgd", batch.norm = FALSE,
                learning.rate=learning.rate, wd=wd, clip_gradient=clip_gradient)`

It turn out an error:
  [23:09:52] d:\chhong\mxnet\src\operator./reshape-inl.h:227: Using target_shape will be deprecated.
[23:09:52] D:\chhong\mxnet\dmlc-core\include\dmlc/logging.h:235: [23:09:52] D:\chhong\mxnet\src\ndarray\ndarray.cc:227: Check failed: from.shape() == to->shape() operands shape mismatch
Error: [23:09:52] D:\chhong\mxnet\src\ndarray\ndarray.cc:227: Check failed: from.shape() == to->shape() operands shape mismatch
How to understand this error message and locate the error code?
Is there any useful tutorial or papers about time series forecasting using rnn or lstm? many thanks!

**Comment by michaeljohannesmeier** [comment]
Hi bifeng, 

i am trying to understand your get.label function..for what do you need it?? this functio is not working..i_d is no found when i try it...

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!
